Check size of uid_t and gid_t using AC_CHECK_SIZEOF

This built-in check is simpler than the previous method and, most
importantly, works when cross-compiling.

Signed-off-by: James Le Cuirot <chewi@aura-online.co.uk>
Signed-off-by: Shrikant Bobade <Shrikant_Bobade@mentor.com>
Index: shadow-4.2.1/configure.in
===================================================================
--- shadow-4.2.1.orig/configure.in	2014-05-09 17:23:10.000000000 +0530
+++ shadow-4.2.1/configure.in	2014-09-11 11:59:00.301507170 +0530
@@ -335,16 +335,10 @@
 	dnl
 	dnl FIXME: check if 32 bit UIDs/GIDs are supported by libc
 	dnl
-	AC_RUN_IFELSE([AC_LANG_SOURCE([
-#include <sys/types.h>
-int main(void) {
-	uid_t u;
-	gid_t g;
-	return (sizeof u < 4) || (sizeof g < 4);
-}
-	])], [id32bit="yes"], [id32bit="no"])
-
-	if test "x$id32bit" = "xyes"; then
+	AC_CHECK_SIZEOF([uid_t],, [#include "sys/types.h"])
+	AC_CHECK_SIZEOF([gid_t],, [#include "sys/types.h"])
+	
+	if test "$ac_cv_sizeof_uid_t" -ge 4 && test "$ac_cv_sizeof_gid_t" -ge 4; then
 		AC_DEFINE(ENABLE_SUBIDS, 1, [Define to support the subordinate IDs.])
 		enable_subids="yes"
 	else
